File: app/src/main/java/com/venditti/placares/adapter/PointsAdapter.java
package com.venditti.placares.adapter;

import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.textfield.TextInputEditText;
import com.venditti.placares.R;
import com.venditti.placares.model.Players;
import com.venditti.placares.model.Points;

import java.util.List;

public class PointsAdapter extends RecyclerView.Adapter<PointsAdapter.MyViewHolder> {

    private List<Players> listJogadores;
    public List<Players> getListJogadores() {
        return listJogadores;
    }

    public PointsAdapter(List<Players> listJogadores){
        this.listJogadores = listJogadores;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View placar = LayoutInflater.from(parent.getContext()).inflate(R.layout.points_layout, parent, false);

        return new MyViewHolder(placar);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        List<Points> listPoints = listJogadores.get(position).getPoints();
        holder.editFez.setText("");
        holder.editFaz.setText("");
        holder.textPlayer.setText(listJogadores.get(position).getName());
        holder.textPontos.setText(listJogadores.get(position).getTotal().toString());

        holder.editFez.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {}
            @Override
            public void afterTextChanged(Editable s) {
                Points points = new Points();
                if(!holder.editFez.getText().toString().isEmpty()) {
                    Integer doing = Integer.valueOf(holder.editFaz.getText().toString());
                    Integer done = Integer.valueOf(holder.editFez.getText().toString());
                    points.setDoing(doing);
                    points.setDone(done);
                    points.setSummation(calculaPontuacao(doing, done));
                }
                listPoints.add(points);
                listJogadores.get(position).setPoints(listPoints);
            }
        });
    }

    private Integer calculaPontuacao(Integer doing, Integer done) {
        if(done.equals(doing))
            return done + 10;
        else
            return done;
    }


    @Override
    public int getItemCount() {
        return listJogadores.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {

        TextInputEditText editFaz, editFez;
        TextView textPlayer, textPontos;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            textPlayer = itemView.findViewById(R.id.textPlayer);
            editFaz = itemView.findViewById(R.id.editFaz);
            editFez = itemView.findViewById(R.id.editFez);
            textPontos = itemView.findViewById(R.id.textPontos);
        }
    }

}

File: app/src/main/java/com/venditti/placares/adapter/PointsAdapter.kt
package com.venditti.placares.adapter

import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.textfield.TextInputEditText
import com.venditti.placares.R
import com.venditti.placares.model.Players
import com.venditti.placares.model.Points

class PointsAdapter(val listJogadores: List<Players?>?, private var rodada: String) : RecyclerView.Adapter<PointsAdapter.MyViewHolder>() {
    fun setRodada(rodada: String) {
        this.rodada = rodada
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val placar = LayoutInflater.from(parent.context).inflate(R.layout.points_layout, parent, false)
        return MyViewHolder(placar)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val listPoints = listJogadores!![holder.adapterPosition]?.points
        holder.editFaz.setText("")
        holder.editFez.setText("")
        holder.textPlayer.text = listJogadores[holder.adapterPosition]?.name
        holder.textPontos.text = listJogadores[holder.adapterPosition]?.total.toString()
        holder.editFez.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
                if (s.toString() == "") listPoints!!.remove(rodada)
            }

            override fun afterTextChanged(s: Editable) {
                val points = Points()
                if (s.toString() != "") {
                    val doing = Integer.valueOf(holder.editFaz.text.toString())
                    val done = Integer.valueOf(holder.editFez.text.toString())
                    points.doing = doing
                    points.done = done
                    points.summation = calculaPontuacao(doing, done)
                    listPoints!![rodada] = points
                    listJogadores[holder.adapterPosition]?.points = listPoints
                }
            }
        })
    }

    private fun calculaPontuacao(doing: Int, done: Int): Int {
        return if (done == doing) done + 10 else done
    }

    override fun getItemCount(): Int {
        return listJogadores!!.size
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var editFaz: TextInputEditText
        var editFez: TextInputEditText
        var textPlayer: TextView
        var textPontos: TextView

        init {
            textPlayer = itemView.findViewById(R.id.textPlayer)
            editFaz = itemView.findViewById(R.id.editFaz)
            editFez = itemView.findViewById(R.id.editFez)
            textPontos = itemView.findViewById(R.id.textPontos)
        }
    }
}